internal/stack/stack_test.go

```diff
@@ -0,0 +1,43 @@
+package stack
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestStack(t *testing.T) {
+	t.Run("test_push_adds_entry_and_creates_new_stack", func(t *testing.T) {
+		firstStack := &Stack[string]{Value: "hello"}
+		secondStack := Push(firstStack, "world")
+
+		require.NotEqual(t, firstStack.Value, secondStack.Value)
+	})
+
+	t.Run("test_pop_does_not_affect_original", func(t *testing.T) {
+		firstStack := &Stack[string]{Value: "hello"}
+
+		val, secondStack := Pop(firstStack)
+		require.Equal(t, "hello", val)
```

Comment:
Would the test makes more sense if we have world as the top item and see if popping the second stack modifies the first?